### PR TITLE
feat(optimizely_fullstack): update implmentation with log endpoint and onboarded on cdk v2

### PR DESCRIPTION
**Pull request by Gauravudia**
## Description of the change

> 
- Removed the transformer implementation using the Agent API, as it requires native implementation at the client's end. (Clients need to host the Agent server and provide the URL in the web app to form the endpoint.)
- Updated the transformer implementation to use the [Log endpoint ](https://docs.developers.optimizely.com/experimentation-data/reference/post_events)for sending data to Optimizely FullStack.
- The Optimizely Agent internally utilizes the [Log endpoint](https://docs.developers.optimizely.com/experimentation-data/reference/post_events) to track events.
- Onboarded the destination on CDK V2.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      - name: mappedEvents
+        template: |
+          const eventsMap = $.getHashFromArray(.destination.Config.pageMapping, 'from', 'to');
+          const category = .message.properties.category.toLowerCase();
+          const name = .message.name.toLowerCase();
+          const mappedEvents = [];
+          .destination.Config.trackCategorizedPages && eventsMap[category] ? mappedEvents.push(eventsMap[category]);
+          .destination.Config.trackNamedPages && eventsMap[name] ? mappedEvents.push(eventsMap[name]);
+          console.log("mappedEvents", mappedEvents);
+          console.log("eventsMap", eventsMap);
```
Comment (ItsSudip) [inline src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml:212]: We can remove these right?

File: src/cdk/v2/destinations/optimizely_fullstack/rtWorkflow.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+bindings:
+  - name: handleRtTfSingleEventError
+    path: ../../../../v0/util/index
+
+steps:
+  - name: validateInput
+    template: |
+      console.log("In ROuter transform....")
```
Comment (ItsSudip) [inline src/cdk/v2/destinations/optimizely_fullstack/rtWorkflow.yaml:8]:
```suggestion
```

File: src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml (hunk 1 above)
Comment (Gauravudia) [reply src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml]: Yes

File: src/cdk/v2/bindings/default.js [hunk 3]
```diff
@@ -27,8 +34,22 @@ function assertConfig(val, message) {
   }
 }
 
+function assertNetwork(response, message) {
```
Comment (koladilip) [inline src/cdk/v2/bindings/default.js:37]: assertNetwork --> assertHttpResp

File: src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+  - name: validateDateFileForTrackPageAndScreen
+    condition: $.outputs.messageType in [ {{$.EventType.TRACK}}, {{$.EventType.PAGE}}, {{$.EventType.SCREEN}}]
+    template: |
+      const availableEvents = $.outputs.dataFile.events;
+      $.assert(Array.isArray(availableEvents) && availableEvents.length > 0, "No events are present in the data file");
+
+  - name: extractExperimentIds
+    template: |
+      const integrationObj = $.getIntegrationsObj(.message, "optimizely_fullstack");
+      {
```
Comment (koladilip) [inline src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml:75]: integrationObj{["campaignId", "experimentId", "variationId"]}

File: src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+          {
+            ...$.outputs.commonPayload,
+            "visitors": $.outputs.preparePageOrScreenPayload.visitors,
+          };
+
+  - name: buildResponseForProcessTransformation
+    description: build response
+    template: |
+      const messageType = $.outputs.messageType;
+      const payload = (messageType === {{$.EventType.IDENTIFY}} ? $.outputs.prepareIdentifyPayload:
```
Comment (koladilip) [inline src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml:302]: const payload = $.outputs.prepareIdentifyPayload || $.outputs.prepareTrackPayload || $.outputs.preparePageOrScreenPayload
Reason: only one of them will be defined right?

**Review by koladilip — DISMISSED** [review]

**Review by utsabc — DISMISSED** [review]

File: src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml [hunk 6]
```diff
@@ -56,9 +57,15 @@ steps:
       "Please enable at one of them to send page/screen events to Optimizely.");
 
   - name: dataFile
+    description: Fetch the data file from the data file url
     template: |
       const dataFileUrl = .destination.Config.dataFileUrl;
-      const rawResponse = await $.handleHttpRequest("get", dataFileUrl);
+      const dataFileAccessToken = .destination.Config.dataFileAccessToken;
+      const rawResponse = .destination.Config.secureEnvironment ?  await $.handleHttpRequest("get", dataFileUrl, {
```
Comment (koladilip) [inline src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml:64]: const options = .destination.Config.secureEnvironment ? {
      headers: {
          "Authorization": "Bearer " + dataFileAccessToken
        }
      };
const rawResponse = await $.handleHttpRequest("get", dataFileUrl, options);

**Review by koladilip — DISMISSED** [review]

**Review by utsabc — DISMISSED** [review]

**Review by utsabc — APPROVED** [review]

**Review by ujjwal-ab — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**